save.js

```diff
@@ -150,4 +150,7 @@ var updateTypes = function() {
 	for( var i = 0; i<player.caughtPokemonList.length; i++) {
 		player.caughtPokemonList[i].type = getPokemonByName(player.caughtPokemonList[i].name).type;
 	}
+	for (i = 0; i < numberToType.length; i++) {
+		player.typeShards[i] = player.typeShards[i] || 0;
+	}
```

Comment:
Replaces all `undefined`, `NaN` and `0` values with `0`.
Applying `+=` or `-=` to undefined 18th value was changing it to `NaN`. This can repair such save.